system_metrics_collector/test/topic_statistics_collector/test_subscriber_topic_statistics.cpp

```diff
@@ -110,20 +116,9 @@ class TestSubscriberTopicStatisticsNode
     return data;
   }
 
-  /**
-   * Spin a MetricsMessageSubscriber node until the desired MetricsMessages are received.
-   */
-  void SpinUntilMessageReceived()
+  std::vector<StatisticData> GetPublishedCollectorData() const
```

Comment:
Return by const reference.

Reply:
the vector is built within the function

Reply:
Why const reference? Won't reads and writes need to be guarded then? Why not just return a copy in order to read data?

Reply:
I think it depends on the intended use case. If the vector is only needed for short periods, then const ref would be preferred. I think the most flexible use would be to return by const ref and then explicitly assign to a non-reference if a copy is required.

Reply:
That's up to the caller.

Even if `GetPublishedCollectorData()` returns by const reference, if the caller wants a copy, it can do:
```
std::vector<StatisticData> data = node->GetPublishedCollectorData();
```
And now if the caller otherwise wants to avoid a copy, do:
```
const std::vector<StatisticData> & data = node->GetPublishedCollectorData();
```

Reply:
Nit:
```c++
auto data = node->GetPublishedCollectorData();
```
`auto` without the `&` will create a local copy.